docs/datadog_ndk_module.md

```diff
@@ -0,0 +1,33 @@
+# Datadog NDK extension package
```

Comment:
```# Android Native Crash Collection```

Also, instead of `datadog_ndk_module.md` rename this file `native_crash_collection.md`